cpp/src/knn/knn_opg_common.cu

```diff
@@ -506,11 +609,9 @@ void opg_knn(raft::handle_t &handle, std::vector<Matrix::Data<T> *> *out,
       device_buffer<float> res_D(allocator, stream);
       if (my_rank_is_idx) {
         /**
-             * All index ranks perform local KNN
-             */
-        if (verbose)
-          std::cout << "Rank " << my_rank << ": Performing Local KNN"
-                    << std::endl;
+         * All index ranks perform local KNN
+         */
+        if (verbose) CUML_LOG_DEBUG("Rank %d: Performing Local KNN", my_rank);
```

Comment:
Thank you for using the logging! We shouldn't need to make this conditional anymore either because the logger does that automatically.